python/ray/util/options.py

```diff
@@ -0,0 +1,206 @@
+    "lifetime": Option(str, options_only=True),
+    "max_concurrency": _counting_option("max_concurrency", False, options_only=True),
+    "max_restarts": _counting_option("max_restarts", default_value=0),
+    "max_task_retries": _counting_option("max_task_retries", default_value=0),
+    "max_pending_calls": _counting_option(
+        "max_pending_calls", options_only=True, default_value=-1
+    ),
+    "namespace": Option((str, type(None)), options_only=True, default_value=None),
+}
+
+# Priority is important here because during dictionary update, same key with higher
+# priority overrides the same key with lower priority. We make use of priority
+# to set the correct default value for tasks / actors.
+
+# priority: _common_options > _actor_only_options > _task_only_options
+valid_options = {**_task_only_options, **_actor_only_options, **_common_options}
+# priority: _task_only_options > _common_options
+task_options = {**_common_options, **_task_only_options}
+# priority: _actor_only_options > _common_options
+actor_options = {**_common_options, **_actor_only_options}
+
+remote_args_error_string = (
+    "The @ray.remote decorator must be applied either with no arguments and no "
+    "parentheses, for example '@ray.remote', or it must be applied using some of "
+    f"the arguments in the list {list(valid_options.keys())}, for example "
+    "'@ray.remote(num_returns=2, resources={\"CustomResource\": 1})'."
+)
+
+
+def validate_task_options(options: Dict[str, Any], include_options_only=True):
+    """Options check for Ray tasks.
+
+    Args:
+        options: Options for Ray tasks.
+        include_options_only: If True, the options that should be only
+            used in ".options()" are included.
+    """
+    for k, v in options.items():
+        if k not in task_options:
+            raise ValueError(
+                f"Invalid option keyword {k} for remote functions. "
+                f"Valid ones are {list(task_options)}."
+            )
+        task_options[k].validate(k, v, include_options_only)
+
+
+def validate_actor_options(options: Dict[str, Any], include_options_only=True):
+    """Options check for Ray actors.
+
+    Args:
+        options: Options for Ray actors.
+        include_options_only: If True, the options that should be only
+            used in ".options()" are included.
+    """
+    for k, v in options.items():
+        if k not in actor_options:
+            raise ValueError(
+                f"Invalid option keyword {k} for actors. "
+                f"Valid ones are {list(actor_options)}."
+            )
+        actor_options[k].validate(k, v, include_options_only)
+
+    if options.get("max_restarts", 0) == 0 and options.get("max_task_retries", 0) != 0:
+        raise ValueError(
+            "'max_task_retries' cannot be set if 'max_restarts' "
+            "is 0 or if 'max_restarts' is not set."
+        )
+
+
+def filter_placeholder_in_option(option_dict: Dict[str, Any]):
```

Comment:
This looks like it is always used with `validate_actor_options ` or `validate_task_options ` except one place.

Is there any way to make this as a private function used in `validate_actor_options` & `validate_task_options`? Or would you like to make them a const method?

Reply:
I removed this function. It is required as a workaround to change less code. Now I cleaned up more code and this is no longer required.